Taken from upstream commit 3bf7fd47707865c3fb50a560adfd71f5038b4c52
"Update Gradle to v9.6.0 (main)"

--- build.gradle.kts.orig	2026-08-07 22:24:57
+++ build.gradle.kts	2026-09-19 16:42:38
@@ -11,13 +11,13 @@
 description = "JUnit"
 group = "org.junit"
 
-val license by extra(License(
+extra["license"] = License(
 	name = "Eclipse Public License v2.0",
 	url = uri("https://www.eclipse.org/legal/epl-v20.html"),
-	headerFile = layout.projectDirectory.file("gradle/config/spotless/eclipse-public-license-2.0.java")
-))
+	headerFile = layout.settingsDirectory.file("gradle/config/spotless/eclipse-public-license-2.0.java")
+)
 
-val platformProjects by extra(listOf(
+val platformProjects = listOf(
 		projects.junitPlatformCommons,
 		projects.junitPlatformConsole,
 		projects.junitPlatformConsoleStandalone,
@@ -28,22 +28,30 @@
 		projects.junitPlatformSuiteApi,
 		projects.junitPlatformSuiteEngine,
 		projects.junitPlatformTestkit
-).map { dependencyProject(it) })
+)
+	.map { dependencyProject(it) }
+	.also { extra["platformProjects"] = it }
 
-val jupiterProjects by extra(listOf(
+val jupiterProjects = listOf(
 		projects.junitJupiter,
 		projects.junitJupiterApi,
 		projects.junitJupiterEngine,
 		projects.junitJupiterMigrationsupport,
 		projects.junitJupiterParams
-).map { dependencyProject(it) })
+)
+	.map { dependencyProject(it) }
+	.also { extra["jupiterProjects"] = it }
 
-val vintageProjects by extra(listOf(
-	dependencyProject(projects.junitVintageEngine)
-))
+val vintageProjects = listOf(
+	projects.junitVintageEngine
+)
+	.map { dependencyProject(it) }
+	.also { extra["vintageProjects"] = it }
 
-val mavenizedProjects by extra(listOf(dependencyProject(projects.junitStart)) + platformProjects + jupiterProjects + vintageProjects)
-val modularProjects by extra(mavenizedProjects - setOf(dependencyProject(projects.junitPlatformConsoleStandalone)))
+val mavenizedProjects = (listOf(dependencyProject(projects.junitStart)) + platformProjects + jupiterProjects + vintageProjects)
+	.also { extra["mavenizedProjects"] = it }
+val modularProjects = (mavenizedProjects - setOf(dependencyProject(projects.junitPlatformConsoleStandalone)))
+	.also { extra["modularProjects"] = it }
 
 dependencies {
 	modularProjects.forEach {
--- documentation/documentation.gradle.kts.orig	2026-08-07 22:24:57
+++ documentation/documentation.gradle.kts	2026-09-19 16:51:38
@@ -19,8 +19,10 @@
 	id("junitbuild.testing-conventions")
 }
 
-val mavenizedProjects: List<Project> by rootProject
-val modularProjects: List<Project> by rootProject
+@Suppress("UNCHECKED_CAST")
+val mavenizedProjects = rootProject.extra["mavenizedProjects"] as List<Project>
+@Suppress("UNCHECKED_CAST")
+val modularProjects = rootProject.extra["modularProjects"] as List<Project>
 
 // Because we need to set up Javadoc aggregation
 modularProjects.forEach { evaluationDependsOn(it.path) }
@@ -50,8 +52,8 @@
 		attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, named("javadoc-since-values"))
 	}
 }
-val tools by sourceSets.creating
-val toolsImplementation by configurations.getting
+val tools = sourceSets.create("tools")
+val toolsImplementation = configurations[tools.implementationConfigurationName]
 
 dependencies {
 	implementation(projects.junitJupiterApi) {
@@ -91,7 +93,7 @@
 	standaloneConsoleLauncher(projects.junitPlatformConsoleStandalone)
 }
 
-val buildRevision: String by rootProject.extra
+val buildRevision = rootProject.extra["buildRevision"] as String
 val snapshot = version.isSnapshot()
 val releaseBranch = if (snapshot) "HEAD" else "r${version}"
 val replaceCurrentDocs = buildParameters.documentation.replaceCurrentDocs
@@ -125,7 +127,7 @@
 	val consoleLauncherTestEventXmlFiles =
 		files(consoleLauncherTestReportsDir.map { it.asFileTree.matching { include("**/open-test-report.xml") } })
 
-	val consoleLauncherTest by registering(RunConsoleLauncher::class) {
+	val consoleLauncherTest = register("consoleLauncherTest", RunConsoleLauncher::class) {
 		args.addAll("execute")
 		args.addAll("--scan-classpath")
 		args.addAll("--config=junit.platform.reporting.open.xml.enabled=true")
@@ -185,35 +187,35 @@
 		config = resources.text.fromFile(checkstyle.configDirectory.file("checkstyleMain.xml"))
 	}
 
-	val generateConsoleLauncherOptions by registering(CaptureJavaExecOutput::class) {
+	val generateConsoleLauncherOptions = register("generateConsoleLauncherOptions", CaptureJavaExecOutput::class) {
 		classpath.from(standaloneConsoleLauncherClasspath)
 		mainClass = "org.junit.platform.console.ConsoleLauncher"
 		args.addAll("--help", "--disable-banner")
 		outputFile = consoleLauncherOptionsFile
 	}
 
-	val generateConsoleLauncherDiscoverOptions by registering(CaptureJavaExecOutput::class) {
+	val generateConsoleLauncherDiscoverOptions = register("generateConsoleLauncherDiscoverOptions", CaptureJavaExecOutput::class) {
 		classpath.from(standaloneConsoleLauncherClasspath)
 		mainClass = "org.junit.platform.console.ConsoleLauncher"
 		args.addAll("discover", "--help", "--disable-banner")
 		outputFile = consoleLauncherDiscoverOptionsFile
 	}
 
-	val generateConsoleLauncherExecuteOptions by registering(CaptureJavaExecOutput::class) {
+	val generateConsoleLauncherExecuteOptions = register("generateConsoleLauncherExecuteOptions", CaptureJavaExecOutput::class) {
 		classpath.from(standaloneConsoleLauncherClasspath)
 		mainClass = "org.junit.platform.console.ConsoleLauncher"
 		args.addAll("execute", "--help", "--disable-banner")
 		outputFile = consoleLauncherExecuteOptionsFile
 	}
 
-	val generateConsoleLauncherEnginesOptions by registering(CaptureJavaExecOutput::class) {
+	val generateConsoleLauncherEnginesOptions = register("generateConsoleLauncherEnginesOptions", CaptureJavaExecOutput::class) {
 		classpath.from(standaloneConsoleLauncherClasspath)
 		mainClass = "org.junit.platform.console.ConsoleLauncher"
 		args.addAll("engines", "--help", "--disable-banner")
 		outputFile = consoleLauncherEnginesOptionsFile
 	}
 
-	val generateApiTables by registering(JavaExec::class) {
+	val generateApiTables = register("generateApiTables", JavaExec::class) {
 		classpath = tools.runtimeClasspath
 		mainClass = "org.junit.api.tools.ApiReportGenerator"
 		systemProperty("api.moduleNames", modularProjects.map { it.javaModuleName }.sorted().joinToString(","))
@@ -229,7 +231,7 @@
 		outputs.file(experimentalApisTableFile)
 	}
 
-	val generateStandaloneConsoleLauncherShadowedArtifactsFile by registering(GenerateStandaloneConsoleLauncherShadowedArtifactsFile::class) {
+	val generateStandaloneConsoleLauncherShadowedArtifactsFile = register("generateStandaloneConsoleLauncherShadowedArtifactsFile", GenerateStandaloneConsoleLauncherShadowedArtifactsFile::class) {
 		inputJar.fileProvider(standaloneConsoleLauncherClasspath.flatMap { it.elements.map { it.single().asFile } })
 		outputFile = standaloneConsoleLauncherShadowedArtifactsFile
 	}
@@ -241,7 +243,7 @@
 
 	val plantUmlOutputDirectory = plantUml.flatMap { it.outputDirectory }
 
-	val generateAsciidocInputs by registering {
+	val generateAsciidocInputs = register("generateAsciidocInputs") {
 		dependsOn(
 			generateConsoleLauncherOptions,
 			generateConsoleLauncherDiscoverOptions,
@@ -253,7 +255,7 @@
 		)
 	}
 
-	val downloadJavadocElementLists by registering {
+	val downloadJavadocElementLists = register("downloadJavadocElementLists") {
 		outputs.cacheIf { true }
 		outputs.dir(elementListsDir).withPropertyName("elementListsDir")
 		inputs.property("externalModulesWithoutModularJavadoc", externalModulesWithoutModularJavadoc)
@@ -268,7 +270,7 @@
 		}
 	}
 
-	val mergeJavadocSinceValues by registering {
+	val mergeJavadocSinceValues = register("mergeJavadocSinceValues") {
 		inputs.files(allJavadocSinceValuesClasspath).withPathSensitivity(PathSensitivity.NONE)
 		val outputFile = layout.buildDirectory.file("docs/aggregated-javadoc-since-values.txt")
 		outputs.file(outputFile)
@@ -289,7 +291,7 @@
 		}
 	}
 
-	val aggregateJavadocs by registering(Javadoc::class) {
+	val aggregateJavadocs = register("aggregateJavadocs", Javadoc::class) {
 		dependsOn(modularProjects.map { it.tasks.jar })
 		dependsOn(downloadJavadocElementLists)
 		group = "Documentation"
@@ -372,7 +374,7 @@
 		options.destinationDirectory = layout.buildDirectory.dir("docs/javadoc").get().asFile
 	}
 
-	val fixJavadoc by registering(Copy::class) {
+	val fixJavadoc = register("fixJavadoc", Copy::class) {
 		dependsOn(aggregateJavadocs)
 		group = "Documentation"
 		description = "Fix links to external API specs in the locally aggregated Javadoc HTML files"
@@ -416,7 +418,7 @@
 		into(layout.buildDirectory.dir("docs/fixedJavadoc"))
 	}
 
-	val prepareGitHubAttestation by registering(Sync::class) {
+	val prepareGitHubAttestation = register("prepareGitHubAttestation", Sync::class) {
 		from(attestationClasspath)
 		into(layout.buildDirectory.dir("attestation"))
 		rename("(.*)-SNAPSHOT.jar", "$1-SNAPSHOT+${buildRevision.substring(0, 7)}.jar")
--- gradle/plugins/antora/src/main/kotlin/junitbuild.antora-conventions.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/antora/src/main/kotlin/junitbuild.antora-conventions.gradle.kts	2026-09-19 16:53:29
@@ -21,7 +21,7 @@
 	dependsOn("generateAntoraYml")
 }
 
-val generateAntoraPlaybook by tasks.registering(Copy::class) {
+val generateAntoraPlaybook = tasks.register("generateAntoraPlaybook", Copy::class) {
 
 	val gitRepoRoot = providers.exec {
 		commandLine("git", "worktree", "list", "--porcelain", "-z")
--- gradle/plugins/backward-compatibility/src/main/kotlin/junitbuild.backward-compatibility.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/backward-compatibility/src/main/kotlin/junitbuild.backward-compatibility.gradle.kts	2026-09-19 16:57:54
@@ -42,7 +42,7 @@
 	}
 }
 
-val downloadPreviousReleaseJar by tasks.registering(Download::class) {
+val downloadPreviousReleaseJar = tasks.register("downloadPreviousReleaseJar", Download::class) {
 	if (gradle.startParameter.isOffline) {
 		enabled = false
 	}
@@ -58,7 +58,7 @@
 
 val roseauCsvFile = layout.buildDirectory.file("reports/roseau/breaking-changes.csv")
 
-val roseau by tasks.registering(RoseauDiff::class) {
+val roseau = tasks.register("roseau", RoseauDiff::class) {
 	if (gradle.startParameter.isOffline) {
 		enabled = false
 	}
@@ -77,7 +77,7 @@
 	reportDir = layout.buildDirectory.dir("reports/roseau")
 }
 
-val checkBackwardCompatibility by tasks.registering {
+val checkBackwardCompatibility = tasks.register("checkBackwardCompatibility") {
 	dependsOn(roseau)
 }
 
--- gradle/plugins/code-generator/src/main/kotlin/junitbuild.code-generator.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/code-generator/src/main/kotlin/junitbuild.code-generator.gradle.kts	2026-09-19 17:00:32
@@ -6,7 +6,7 @@
 	java
 }
 
-val templates by sourceSets.creating
+val templates = sourceSets.create("templates")
 val templatesCompileOnly = configurations[templates.compileOnlyConfigurationName]
 
 dependencies {
@@ -14,10 +14,10 @@
 	templatesCompileOnly("junitbuild.base:code-generator-model")
 }
 
-val license: License by rootProject.extra
+val license = rootProject.extra["license"] as License
 val rootTargetDir = layout.buildDirectory.dir("generated/sources/jte")
 
-val generateCode by tasks.registering {
+val generateCode = tasks.register("generateCode") {
 	dependsOn(tasks.withType<GenerateJreRelatedSourceCode>())
 	group = LifecycleBasePlugin.BUILD_GROUP
 	description = "Generates JRE-related source code."
--- gradle/plugins/common/src/main/kotlin/junitbuild.build-metadata.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/common/src/main/kotlin/junitbuild.build-metadata.gradle.kts	2026-09-19 17:02:07
@@ -25,10 +25,8 @@
 	.orNull
 	?: OffsetDateTime.now()
 
-val buildDate: String by extra { dateFormatter.format(buildTimeAndDate) }
-val buildTime: String by extra { timeFormatter.format(buildTimeAndDate) }
-val buildRevision: String by extra {
-	providers.exec {
-		commandLine("git", "rev-parse", "--verify", "HEAD")
-	}.standardOutput.asText.get().trim()
-}
+extra["buildDate"] = dateFormatter.format(buildTimeAndDate)
+extra["buildTime"] = timeFormatter.format(buildTimeAndDate)
+extra["buildRevision"] = providers.exec {
+	commandLine("git", "rev-parse", "--verify", "HEAD")
+}.standardOutput.asText.get().trim()
--- gradle/plugins/common/src/main/kotlin/junitbuild.eclipse-conventions.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/common/src/main/kotlin/junitbuild.eclipse-conventions.gradle.kts	2026-09-19 17:03:29
@@ -16,6 +16,8 @@
 	jdt {
 		sourceCompatibility = JavaVersion.VERSION_25
 		targetCompatibility = JavaVersion.VERSION_25
+		// Disable deprecation warning for now (will probably have to removed for Gradle 10)
+		@Suppress("DEPRECATION")
 		file {
 			// Set properties for org.eclipse.jdt.core.prefs
 			withProperties {
--- gradle/plugins/common/src/main/kotlin/junitbuild.jacoco-aggregation-conventions.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/common/src/main/kotlin/junitbuild.jacoco-aggregation-conventions.gradle.kts	2026-09-19 17:04:45
@@ -3,7 +3,7 @@
 	`jacoco-report-aggregation`
 }
 
-val jacocoRootReport by reporting.reports.creating(JacocoCoverageReport::class) {
+val jacocoRootReport = reporting.reports.create("jacocoRootReport", JacocoCoverageReport::class) {
 	testSuiteName = "test"
 }
 
@@ -17,7 +17,7 @@
 
 tasks {
 	val reportTask = named<JacocoReport>(jacocoRootReport.reportTask.name)
-	val jacocoRootCoverageVerification by registering(JacocoCoverageVerification::class) {
+	val jacocoRootCoverageVerification = register("jacocoRootCoverageVerification", JacocoCoverageVerification::class) {
 		enabled = !buildParameters.junit.develocity.predictiveTestSelection.enabled
 		executionData.from(reportTask.map { it.executionData })
 		classDirectories.from(reportTask.map { it.classDirectories })
--- gradle/plugins/common/src/main/kotlin/junitbuild.jacoco-java-conventions.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/common/src/main/kotlin/junitbuild.jacoco-java-conventions.gradle.kts	2026-09-19 17:06:43
@@ -7,7 +7,8 @@
 	id("junitbuild.jacoco-conventions")
 }
 
-val mavenizedProjects: List<Project> by rootProject.extra
+@Suppress("UNCHECKED_CAST")
+val mavenizedProjects = rootProject.extra["mavenizedProjects"] as List<Project>
 
 tasks.withType<Test>().configureEach {
 	configure<JacocoTaskExtension> {
@@ -15,7 +16,7 @@
 	}
 }
 
-val codeCoverageClassesJar by tasks.registering(Jar::class) {
+val codeCoverageClassesJar = tasks.register("codeCoverageClassesJar", Jar::class) {
 	from(tasks.jar.map { zipTree(it.archiveFile) })
 	archiveClassifier = "jacoco"
 	enabled = project in mavenizedProjects
--- gradle/plugins/common/src/main/kotlin/junitbuild.java-library-conventions.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/common/src/main/kotlin/junitbuild.java-library-conventions.gradle.kts	2026-09-19 17:08:00
@@ -11,10 +11,11 @@
 	id("junitbuild.java-errorprone-conventions")
 }
 
-val mavenizedProjects: List<Project> by rootProject.extra
-val buildDate: String by rootProject.extra
-val buildTime: String by rootProject.extra
-val buildRevision: Any by rootProject.extra
+@Suppress("UNCHECKED_CAST")
+val mavenizedProjects = rootProject.extra["mavenizedProjects"] as List<Project>
+val buildDate = rootProject.extra["buildDate"] as String
+val buildTime = rootProject.extra["buildTime"] as String
+val buildRevision = rootProject.extra["buildRevision"] as String
 
 val extension = extensions.create<JavaLibraryExtension>("javaLibrary")
 
--- gradle/plugins/common/src/main/kotlin/junitbuild.javadoc-conventions.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/common/src/main/kotlin/junitbuild.javadoc-conventions.gradle.kts	2026-09-19 17:08:41
@@ -69,7 +69,7 @@
 	}
 }
 
-val extractJavadocSinceValues by tasks.registering {
+val extractJavadocSinceValues = tasks.register("extractJavadocSinceValues") {
 	inputs.files(sourceSets.main.get().allJava).withPathSensitivity(PathSensitivity.NONE)
 	val outputFile = layout.buildDirectory.file("docs/javadoc-since-values.txt")
 	outputs.file(outputFile)
--- gradle/plugins/common/src/main/kotlin/junitbuild.junit4-compatibility.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/common/src/main/kotlin/junitbuild.junit4-compatibility.gradle.kts	2026-09-19 17:09:34
@@ -24,8 +24,8 @@
 }
 
 tasks {
-	val test_4_12 by registering(Test::class) {
-		val test by testing.suites.existing(JvmTestSuite::class)
+	val test_4_12 = register("test_4_12", Test::class) {
+		val test = testing.suites.named<JvmTestSuite>("test")
 		testClassesDirs = files(test.map { it.sources.output.classesDirs })
 		classpath = files(sourceSets.main.map { it.output }) + files(test.map { it.sources.output }) + junit_4_12_classpath.get()
 	}
--- gradle/plugins/common/src/main/kotlin/junitbuild.osgi-conventions.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/common/src/main/kotlin/junitbuild.osgi-conventions.gradle.kts	2026-09-19 17:11:48
@@ -14,9 +14,12 @@
 	it == "jar" || it == "shadowJar"
 }.all { // configure tasks eagerly as workaround for https://github.com/bndtools/bnd/issues/5695
 
-	val importAPIGuardian by extra { "org.apiguardian.*;resolution:=\"optional\"" }
-	val importJSpecify by extra { "org.jspecify.*;resolution:=\"optional\"" }
-	val importCommonsLogging by extra { "org.junit.platform.commons.logging;status=INTERNAL" }
+	val importAPIGuardian = "org.apiguardian.*;resolution:=\"optional\""
+		.also { extra["importAPIGuardian"] = it }
+	val importJSpecify = "org.jspecify.*;resolution:=\"optional\""
+		.also { extra["importJSpecify"] = it }
+	val importCommonsLogging = "org.junit.platform.commons.logging;status=INTERNAL"
+		.also { extra["importCommonsLogging"] = it }
 
 	extensions.create<BundleTaskExtension>(BundleTaskExtension.NAME, this).apply {
 		properties.set(projectDescription.map {
@@ -80,7 +83,7 @@
 // Bnd's Resolve task uses a properties file for its configuration. This
 // task writes out the properties necessary for it to verify the OSGi
 // metadata.
-val osgiProperties by tasks.registering(WriteProperties::class) {
+val osgiProperties = tasks.register("osgiProperties", WriteProperties::class) {
 	destinationFile = layout.buildDirectory.file("verifyOSGiProperties.bndrun")
 	property("-standalone", true)
 	project.extensions.getByType(JavaLibraryExtension::class).let { javaLibrary ->
@@ -102,7 +105,7 @@
 // Bnd's Resolve task is what verifies that a jar can be used in OSGi and
 // that its metadata is valid. If the metadata is invalid this task will
 // fail.
-val verifyOSGi by tasks.registering(Resolve::class) {
+val verifyOSGi = tasks.register("verifyOSGi", Resolve::class) {
 	bndrun = osgiProperties.flatMap { it.destinationFile }
 	outputBndrun = layout.buildDirectory.file("resolvedOSGiProperties.bndrun")
 	isReportOptional = false
--- gradle/plugins/common/src/main/kotlin/junitbuild.publishing-conventions.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/common/src/main/kotlin/junitbuild.publishing-conventions.gradle.kts	2026-09-19 17:13:03
@@ -7,10 +7,15 @@
 	id("junitbuild.build-parameters")
 }
 
-val jupiterProjects: List<Project> by rootProject
-val platformProjects: List<Project> by rootProject
-val vintageProjects: List<Project> by rootProject
+@Suppress("UNCHECKED_CAST")
+val jupiterProjects = rootProject.extra["jupiterProjects"] as List<Project>
 
+@Suppress("UNCHECKED_CAST")
+val platformProjects = rootProject.extra["platformProjects"] as List<Project>
+
+@Suppress("UNCHECKED_CAST")
+val vintageProjects = rootProject.extra["vintageProjects"] as List<Project>
+
 group = buildParameters.publishing.group
 	.getOrElse(when (project) {
 		in jupiterProjects -> "org.junit.jupiter"
@@ -49,7 +54,7 @@
 				}
 				licenses {
 					license {
-						val license: License by rootProject.extra
+						val license = rootProject.extra["license"] as License
 						name = license.name
 						url = license.url.toString()
 					}
--- gradle/plugins/common/src/main/kotlin/junitbuild.spotless-conventions.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/common/src/main/kotlin/junitbuild.spotless-conventions.gradle.kts	2026-09-19 17:14:07
@@ -5,8 +5,6 @@
 	id("com.diffplug.spotless")
 }
 
-val license: License by rootProject.extra
-
 spotless {
 
 	format("misc") {
@@ -25,6 +23,7 @@
 
 	pluginManager.withPlugin("java") {
 
+		val license = rootProject.extra["license"] as License
 		val configDir = rootProject.layout.projectDirectory.dir("gradle/config/eclipse")
 		val importOrderConfigFile = configDir.file("junit-eclipse.importorder")
 		val javaFormatterConfigFile = configDir.file("junit-eclipse-formatter-settings.xml")
@@ -53,6 +52,7 @@
 	}
 
 	pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
+		val license = rootProject.extra["license"] as License
 		kotlin {
 			targetExclude("**/src/test/resources/**")
 			ktlint(requiredVersionFromLibs("ktlint"))
@@ -64,6 +64,7 @@
 	}
 
 	pluginManager.withPlugin("groovy") {
+		val license = rootProject.extra["license"] as License
 		groovy {
 			licenseHeaderFile(license.headerFile)
 			trimTrailingWhitespace()
--- gradle/plugins/common/src/main/kotlin/junitbuild.testing-conventions.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/common/src/main/kotlin/junitbuild.testing-conventions.gradle.kts	2026-09-19 17:15:12
@@ -37,7 +37,7 @@
 	}
 }
 
-val generateOpenTestHtmlReport by tasks.registering(JavaExec::class) {
+val generateOpenTestHtmlReport = tasks.register("generateOpenTestHtmlReport", JavaExec::class) {
 	mustRunAfter(tasks.withType<Test>())
 	mainModule.set("org.opentest4j.reporting.cli")
 	modularity.inferModulePath = true
--- gradle/plugins/publishing/src/main/kotlin/junitbuild.temp-maven-repo.gradle.kts.orig	2026-08-07 22:24:57
+++ gradle/plugins/publishing/src/main/kotlin/junitbuild.temp-maven-repo.gradle.kts	2026-09-19 17:17:43
@@ -1,19 +1,20 @@
 import junitbuild.extensions.capitalized
 import junitbuild.release.VerifyBinaryArtifactsAreIdentical
+import org.gradle.internal.extensions.core.extra
 
-val tempRepoName by extra("temp")
-val tempRepoDir by extra {
-	layout.buildDirectory.dir("repo").get().asFile
-}
+val tempRepoName = "temp"
+val tempRepoDir = layout.buildDirectory.dir("repo").get().asFile
+extra["tempRepoName"] = tempRepoName
+extra["tempRepoDir"] = tempRepoDir
 
-val clearTempRepoDir by tasks.registering {
+val clearTempRepoDir = tasks.register("clearTempRepoDir") {
 	val dir = tempRepoDir
 	doFirst {
 		dir.deleteRecursively()
 	}
 }
 
-val publishAllSubprojectsToTempRepository by tasks.registering
+val publishAllSubprojectsToTempRepository = tasks.register("publishAllSubprojectsToTempRepository")
 
 tasks.register<VerifyBinaryArtifactsAreIdentical>("verifyArtifactsInStagingRepositoryAreReproducible") {
 	dependsOn(publishAllSubprojectsToTempRepository)
--- gradle/wrapper/gradle-wrapper.properties.orig	2026-08-07 22:24:57
+++ gradle/wrapper/gradle-wrapper.properties	2026-09-19 17:18:54
@@ -1,7 +1,7 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionSha256Sum=bafc141b619ad6350fd975fc903156dd5c151998cc8b058e8c1044ab5f7b031f
-distributionUrl=https\://services.gradle.org/distributions/gradle-9.5.1-bin.zip
+distributionSha256Sum=bbaeb2fef8710818cf0e261201dab964c572f92b942812df0c3620d62a529a01
+distributionUrl=https\://services.gradle.org/distributions/gradle-9.6.0-bin.zip
 networkTimeout=10000
 retries=0
 retryBackOffMs=500
--- junit-bom/junit-bom.gradle.kts.orig	2026-08-07 22:24:57
+++ junit-bom/junit-bom.gradle.kts	2026-09-19 17:20:29
@@ -7,7 +7,8 @@
 
 dependencies {
 	constraints {
-		val mavenizedProjects: List<Project> by rootProject.extra
+		@Suppress("UNCHECKED_CAST")
+		val mavenizedProjects = rootProject.extra["mavenizedProjects"] as List<Project>
 		mavenizedProjects.sorted()
 				.filter { it.name != "junit-platform-console-standalone" }
 				.forEach {
--- junit-jupiter-api/junit-jupiter-api.gradle.kts.orig	2026-08-07 22:24:57
+++ junit-jupiter-api/junit-jupiter-api.gradle.kts	2026-09-19 17:20:59
@@ -48,7 +48,7 @@
 			""")
 		}
 	}
-	val generateJreTestDouble by registering(GenerateJreRelatedSourceCode::class) {
+	val generateJreTestDouble = register("generateJreTestDouble", GenerateJreRelatedSourceCode::class) {
 		templateDir = layout.projectDirectory.dir("src/templates/resources/main")
 		targetDir = layout.buildDirectory.dir("generated/sources/jte/testDouble")
 		maxVersion = 22
--- junit-jupiter-migrationsupport/junit-jupiter-migrationsupport.gradle.kts.orig	2026-08-07 22:24:57
+++ junit-jupiter-migrationsupport/junit-jupiter-migrationsupport.gradle.kts	2026-09-19 17:22:21
@@ -23,15 +23,12 @@
 	}
 	jar {
 		bundle {
-			val importAPIGuardian: String by extra
-			val importJSpecify: String by extra
-			val importCommonsLogging: String by extra
 			bnd("""
 				# Import JUnit4 packages with a version
 				Import-Package: \
-					$importAPIGuardian,\
-					$importJSpecify,\
-					$importCommonsLogging,\
+					${extra["importAPIGuardian"]},\
+					${extra["importJSpecify"]},\
+					${extra["importCommonsLogging"]},\
 					org.junit;version="[${libs.versions.junit4Min.get()},5)",\
 					org.junit.rules;version="[${libs.versions.junit4Min.get()},5)",\
 					*
--- junit-jupiter-params/junit-jupiter-params.gradle.kts.orig	2026-08-07 22:24:57
+++ junit-jupiter-params/junit-jupiter-params.gradle.kts	2026-09-19 17:22:52
@@ -38,7 +38,7 @@
 			""")
 		}
 	}
-	val extractFastCSVLicense by registering(Sync::class) {
+	val extractFastCSVLicense = register("extractFastCSVLicense", Sync::class) {
 		from(zipTree(configurations.shadowedClasspath.flatMap { it.elements }.map { it.single { file -> file.asFile.name.contains("fastcsv") } })) {
 			include("META-INF/LICENSE")
 			rename { "LICENSE-fastcsv" }
--- junit-platform-commons/junit-platform-commons.gradle.kts.orig	2026-08-07 22:24:57
+++ junit-platform-commons/junit-platform-commons.gradle.kts	2026-09-19 17:24:21
@@ -39,12 +39,10 @@
 
 tasks.jar {
 	bundle {
-		val importAPIGuardian: String by extra
-		val importJSpecify: String by extra
 		bnd("""
 			Import-Package: \
-				$importAPIGuardian,\
-				$importJSpecify,\
+				${extra["importAPIGuardian"]},\
+				${extra["importJSpecify"]},\
 				kotlin.*;resolution:="optional",\
 				kotlinx.*;resolution:="optional",\
 				*
--- junit-platform-console-standalone/junit-platform-console-standalone.gradle.kts.orig	2026-08-07 22:24:57
+++ junit-platform-console-standalone/junit-platform-console-standalone.gradle.kts	2026-09-19 17:25:40
@@ -33,11 +33,11 @@
 			attributes("Main-Class" to "org.junit.platform.console.ConsoleLauncher")
 		}
 	}
-	val shadowedArtifactsFile by registering(WriteArtifactsFile::class) {
+	val shadowedArtifactsFile = register("shadowedArtifactsFile", WriteArtifactsFile::class) {
 		from(configurations.shadowedClasspath)
 		outputFile = layout.buildDirectory.file("shadowed-artifacts")
 	}
-	val extractThirdPartyLicenses by registering(Sync::class) {
+	val extractThirdPartyLicenses = register("extractThirdPartyLicenses", Sync::class) {
 		from(withArchiveOperations { ops -> configurations.shadowedClasspath.flatMap { it.elements }.map { it.map(ops::zipTree) } })
 		into(layout.buildDirectory.dir("thirdPartyLicenses"))
 		include("LICENSE.txt")
@@ -71,13 +71,11 @@
 		}
 
 		bundle {
-			val importAPIGuardian: String by extra
-			val importJSpecify: String by extra
 			bnd("""
 				# Customize the imports because this is an aggregate jar
 				Import-Package: \
-					$importAPIGuardian,\
-					$importJSpecify,\
+					${extra["importAPIGuardian"]},\
+					${extra["importJSpecify"]},\
 					kotlin.*;resolution:="optional",\
 					kotlinx.*;resolution:="optional",\
 					*
--- junit-platform-launcher/junit-platform-launcher.gradle.kts.orig	2026-08-07 22:24:57
+++ junit-platform-launcher/junit-platform-launcher.gradle.kts	2026-09-19 17:28:02
@@ -22,21 +22,17 @@
 tasks {
 	jar {
 		bundle {
-			val importAPIGuardian: String by extra
-			val importJSpecify: String by extra
-			val importCommonsLogging: String by extra
-			val version = project.version
 			bnd("""
 				Import-Package: \
-					${importAPIGuardian},\
-					${importJSpecify},\
-					${importCommonsLogging},\
+					${extra["importAPIGuardian"]},\
+					${extra["importJSpecify"]},\
+					${extra["importCommonsLogging"]},\
 					jdk.jfr;resolution:="optional",\
 					*
 				Provide-Capability:\
 					org.junit.platform.launcher;\
 						org.junit.platform.launcher='junit-platform-launcher';\
-						version:Version="${'$'}{version_cleanup;${version}}"
+						version:Version="${'$'}{version_cleanup;${project.version}}"
 			""")
 		}
 	}
--- junit-vintage-engine/junit-vintage-engine.gradle.kts.orig	2026-08-07 22:24:57
+++ junit-vintage-engine/junit-vintage-engine.gradle.kts	2026-09-19 17:29:33
@@ -45,15 +45,12 @@
 		bundle {
 			val junit4Min = libs.versions.junit4Min.get()
 			val version = project.version
-			val importAPIGuardian: String by extra
-			val importJSpecify: String by extra
-			val importCommonsLogging: String by extra
 			bnd("""
 				# Import JUnit4 packages with a version
 				Import-Package: \
-					${importAPIGuardian},\
-					${importJSpecify},\
-					${importCommonsLogging},\
+					${extra["importAPIGuardian"]},\
+					${extra["importJSpecify"]},\
+					${extra["importCommonsLogging"]},\
 					junit.runner;version="[${junit4Min},5)",\
 					org.junit;version="[${junit4Min},5)",\
 					org.junit.experimental.categories;version="[${junit4Min},5)",\
@@ -73,8 +70,8 @@
 			""")
 		}
 	}
-	val testWithoutJUnit4 by registering(Test::class) {
-		val test by testing.suites.existing(JvmTestSuite::class)
+	val testWithoutJUnit4 = register("testWithoutJUnit4", Test::class) {
+		val test = testing.suites.named<JvmTestSuite>("test")
 		(options as JUnitPlatformOptions).apply {
 			includeTags("missing-junit4")
 		}
--- platform-tests/platform-tests.gradle.kts.orig	2026-08-07 22:24:57
+++ platform-tests/platform-tests.gradle.kts	2026-09-19 17:32:21
@@ -11,15 +11,15 @@
 	id("junitbuild.jmh-conventions")
 }
 
-val processStarter by sourceSets.creating {
+val sourceSet = sourceSets.create("processStarter") {
 	java {
 		srcDir("src/processStarter/java")
 	}
 }
 
 java {
-	registerFeature(processStarter.name) {
-		usingSourceSet(processStarter)
+	registerFeature(sourceSet.name) {
+		usingSourceSet(sourceSet)
 	}
 }
 
@@ -64,7 +64,8 @@
 	}
 
 	// --- Test run-time dependencies ---------------------------------------------
-	val mavenizedProjects: List<Project> by rootProject
+	@Suppress("UNCHECKED_CAST")
+	val mavenizedProjects = rootProject.extra["mavenizedProjects"] as List<Project>
 	mavenizedProjects.filter { it.path != projects.junitPlatformConsoleStandalone.path }.forEach {
 		// Add all projects to the classpath for tests using classpath scanning
 		testRuntimeOnly(it)
@@ -79,13 +80,13 @@
 	jmh(libs.junit4)
 
 	// --- ProcessStarter dependencies --------------------------------------------
-	processStarter.implementationConfigurationName(libs.groovy) {
+	sourceSet.implementationConfigurationName(libs.groovy) {
 		because("it provides convenience methods to handle process output")
 	}
-	processStarter.implementationConfigurationName(libs.commons.io) {
+	sourceSet.implementationConfigurationName(libs.commons.io) {
 		because("it uses TeeOutputStream")
 	}
-	processStarter.implementationConfigurationName(libs.opentest4j) {
+	sourceSet.implementationConfigurationName(libs.opentest4j) {
 		because("it throws TestAbortedException")
 	}
 }
@@ -120,8 +121,8 @@
 			includeTags("junit4")
 		}
 	}
-	val testWoodstox by registering(Test::class) {
-		val test by testing.suites.existing(JvmTestSuite::class)
+	val testWoodstox = register("testWoodstox", Test::class) {
+		val test = testing.suites.named<JvmTestSuite>("test")
 		testClassesDirs = files(test.map { it.sources.output.classesDirs })
 		classpath = files(sourceSets.main.map { it.output }) + files(test.map { it.sources.output }) + woodstoxRuntimeClasspath.get()
 		group = JavaBasePlugin.VERIFICATION_GROUP
@@ -130,10 +131,10 @@
 	check {
 		dependsOn(testWoodstox)
 	}
-	named<JavaCompile>(processStarter.compileJavaTaskName).configure {
+	named<JavaCompile>(sourceSet.compileJavaTaskName).configure {
 		options.release = javaLibrary.testJavaVersion.map { it.majorVersion.toInt() }
 	}
-	named<Checkstyle>("checkstyle${processStarter.name.capitalized()}").configure {
+	named<Checkstyle>("checkstyle${sourceSet.name.capitalized()}").configure {
 		config = resources.text.fromFile(checkstyle.configDirectory.file("checkstyleMain.xml"))
 	}
 }
--- platform-tooling-support-tests/platform-tooling-support-tests.gradle.kts.orig	2026-08-07 22:24:57
+++ platform-tooling-support-tests/platform-tooling-support-tests.gradle.kts	2026-09-19 17:36:27
@@ -52,7 +52,8 @@
 	extendsFrom(mavenDistribution.get())
 }
 
-val modularProjects: List<Project> by rootProject
+@Suppress("UNCHECKED_CAST")
+val modularProjects = rootProject.extra["modularProjects"] as List<Project>
 
 dependencies {
 	implementation(libs.commons.io) {
@@ -101,16 +102,17 @@
 
 val mavenDistributionDir = layout.buildDirectory.dir("maven-distribution")
 
-val unzipMavenDistribution by tasks.registering(Sync::class) {
+val unzipMavenDistribution = tasks.register("unzipMavenDistribution", Sync::class) {
 	from(zipTree(mavenDistributionClasspath.flatMap { d -> d.elements.map { e -> e.single() } }))
 	into(mavenDistributionDir)
 }
 
-val normalizeMavenRepo by tasks.registering(Sync::class) {
+val normalizeMavenRepo = tasks.register("normalizeMavenRepo", Sync::class) {
 
-	val mavenizedProjects: List<Project> by rootProject
-	val tempRepoDir: File by rootProject
-	val tempRepoName: String by rootProject
+	@Suppress("UNCHECKED_CAST")
+	val mavenizedProjects = rootProject.extra["mavenizedProjects"] as List<Project>
+	val tempRepoDir = rootProject.extra["tempRepoDir"] as File
+	val tempRepoName = rootProject.extra["tempRepoName"] as String
 
 	// All maven-aware projects must be published to the local temp repository
 	(mavenizedProjects + dependencyProject(projects.junitBom))
@@ -132,7 +134,7 @@
 	into(layout.buildDirectory.dir("normalized-repo"))
 }
 
-val archUnit by testing.suites.registering(JvmTestSuite::class) {
+val archUnit = testing.suites.register("archUnit", JvmTestSuite::class) {
 	dependencies {
 		implementation(libs.archunit) {
 			because("checking the architecture")
@@ -212,7 +214,7 @@
 	dependsOn(archUnit, graalVmTest)
 }
 
-val test by testing.suites.getting(JvmTestSuite::class) {
+testing.suites.named<JvmTestSuite>("test") {
 	dependencies {
 		implementation(libs.bndlib) {
 			because("parsing OSGi metadata")
--- platform-tooling-support-tests/projects/gradle-kotlin-extensions/build.gradle.kts.orig	2026-08-07 22:24:57
+++ platform-tooling-support-tests/projects/gradle-kotlin-extensions/build.gradle.kts	2026-09-19 17:41:37
@@ -11,7 +11,7 @@
 	mavenCentral()
 }
 
-val junitVersion: String by project
+val junitVersion = providers.gradleProperty("junitVersion").orNull
 
 dependencies {
 	testImplementation(kotlin("stdlib"))
--- platform-tooling-support-tests/projects/gradle-missing-engine/build.gradle.kts.orig	2026-08-07 22:24:57
+++ platform-tooling-support-tests/projects/gradle-missing-engine/build.gradle.kts	2026-09-19 17:42:14
@@ -2,7 +2,7 @@
 	java
 }
 
-val junitVersion: String by project
+val junitVersion = providers.gradleProperty("junitVersion").orNull
 
 repositories {
 	maven { url = uri(file(System.getProperty("maven.repo"))) }
--- platform-tooling-support-tests/projects/jupiter-starter/build.gradle.kts.orig	2026-08-07 22:24:57
+++ platform-tooling-support-tests/projects/jupiter-starter/build.gradle.kts	2026-09-19 17:42:50
@@ -2,7 +2,7 @@
 	java
 }
 
-val junitVersion: String by project
+val junitVersion = providers.gradleProperty("junitVersion").orNull
 
 repositories {
 	maven { url = uri(file(System.getProperty("maven.repo"))) }
--- platform-tooling-support-tests/projects/kotlin-coroutines/build.gradle.kts.orig	2026-08-07 22:24:57
+++ platform-tooling-support-tests/projects/kotlin-coroutines/build.gradle.kts	2026-09-19 17:43:21
@@ -2,7 +2,7 @@
 	kotlin("jvm") version "2.3.21"
 }
 
-val junitVersion: String by project
+val junitVersion = providers.gradleProperty("junitVersion").orNull
 
 repositories {
 	maven { url = uri(file(System.getProperty("maven.repo"))) }
--- platform-tooling-support-tests/projects/reflection-tests/build.gradle.kts.orig	2026-08-07 22:24:57
+++ platform-tooling-support-tests/projects/reflection-tests/build.gradle.kts	2026-09-19 17:43:49
@@ -2,7 +2,7 @@
 	java
 }
 
-val junitVersion: String by project
+val junitVersion = providers.gradleProperty("junitVersion").orNull
 
 repositories {
 	maven { url = uri(file(System.getProperty("maven.repo"))) }
--- platform-tooling-support-tests/projects/vintage/build.gradle.kts.orig	2026-08-07 22:24:57
+++ platform-tooling-support-tests/projects/vintage/build.gradle.kts	2026-09-19 17:44:15
@@ -9,7 +9,7 @@
 	mavenCentral()
 }
 
-val junitVersion: String by project
+val junitVersion = providers.gradleProperty("junitVersion").orNull
 
 dependencies {
 	val junit4Version = System.getProperty("junit4Version", "4.12")
--- settings.gradle.kts.orig	2026-08-07 22:24:57
+++ settings.gradle.kts	2026-09-19 17:58:30
@@ -105,3 +105,4 @@
 
 enableFeaturePreview("STABLE_CONFIGURATION_CACHE")
 enableFeaturePreview("TYPESAFE_PROJECT_ACCESSORS")
+enableFeaturePreview("NO_IMPLICIT_LOOKUP_IN_PARENT_PROJECTS")
